src/devices_models/devices/common/add_variable.jl

```diff
@@ -141,25 +65,79 @@ If binary = true:
 """
 function add_variable!(
     psi_container::PSIContainer,
-    devices::D,
-    var_name::Symbol,
-    binary::Bool,
-    expression_name::Union{Nothing, Symbol} = nothing,
-    sign::Float64 = 1.0;
-    kwargs...,
-) where {D <: Union{Vector{<:PSY.Component}, IS.FlattenIteratorWrapper{<:PSY.Component}}}
+    variable_type::VariableType,
+    devices::U,
+) where {U <: Union{Vector{D}, IS.FlattenIteratorWrapper{D}}} where D<:PSY.Component
+
     @assert !isempty(devices)
     time_steps = model_time_steps(psi_container)
+
+    var_name = make_variable_name(typeof(variable_type), eltype(devices))
+    binary = get_variable_binary(variable_type, eltype(devices))
+    expression_name = get_variable_expression_name(variable_type, eltype(devices))
+    sign = get_variable_sign(variable_type, eltype(devices))
+
     variable = add_var_container!(
         psi_container,
         var_name,
         [PSY.get_name(d) for d in devices],
         time_steps,
     )
 
-    lb_f = get(kwargs, :lb_value, nothing)
-    init_f = get(kwargs, :initial_value, nothing)
-    ub_f = get(kwargs, :ub_value, nothing)
+    for t in time_steps, d in devices
+        name = PSY.get_name(d)
+        variable[name, t] = JuMP.@variable(
+            psi_container.JuMPmodel,
+            base_name = "$(var_name)_{$(name), $(t)}",
+            binary = binary
+        )
+
+        ub = get_variable_upper_bound(variable_type, d, psi_container.settings)
+        !isnothing(ub) && JuMP.set_upper_bound(variable[name, t], ub)
```

Comment:
Makes sense. Would you like me to make these changes in this PR? Current status of the codebase is as follows:

```plaintext
$ rg isnothing src --count-matches | wc -l
23

$ rg '=== nothing' src --count-matches | wc -l
2

$ rg '!== nothing' src --count-matches | wc -l
0
```

Reply:
I personally started writing `===` instead of `isnothing` but didn't sweep through the codebases and change all occurrences. I also don't know if Jose came to the same conclusion.

Reply:
making the change in this PR is appropriate.

Reply:
Resolved in https://github.com/NREL-SIIP/PowerSimulations.jl/pull/560/commits/7fb5eda67b7873f612ac8be80e7f358bea57c2ca.